opcodes/dxil/dxil_resources.cpp

```diff
@@ -323,6 +323,10 @@ bool emit_store_output_instruction(Converter::Impl &impl, const llvm::CallInst *
 	if (!get_constant_operand(instruction, 1, &output_element_index))
 		return false;
 
+	// Mask out writes to unused higher RTs when using dual source blending.
+	if (impl.options.dual_source_blending && output_element_index >= 2)
```

Comment:
This is not necessarily correct. The output element index needs to index to impl.output_elements_meta[] as below. It's in there that it is known what the start row is. It's possible that information needs to be added.

Reply:
Is there a test that could catch a situation where the output element index is not sufficient and the meta element is?

Reply:
It would likely only show up in hand-written LLVM asm. It might also show up if you reorder the output arguments so it's SV_Target2, SV_Target1, etc.